doc/acl.md

```diff
@@ -1 +1,762 @@
+    "status_message": "OK",
+     "data": {
+        "id":3,
+        "parent_id": 2,
+        "alias": "Groups:index",
+        "created": "2020-10-28T08:58:13+00:00",
+        "modified": "2020-10-28T08:58:13+00:00"
+    }
+}
+```
+
+### Logic
+ using <a href="https://book.cakephp.org/4/en/orm/behaviors/tree.html" rel="notfollow">cakephp tree</a> behavior to get lft and rght from parent_id
+
+#### Validation
+- parent_id: required and not empty
+- alias: required, not empty and unique
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "parent_id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "parent_id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "parent_id": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "Parent_id is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload :
+```
+{
+    "parent_id": 2
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "alias is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+Request Payload :
+```
+{
+    "parent_id": 2,
+    "alias": ""
+}
+```
+ 
+Response HTTP Status Code : 400
+
+Response Payload:
+```
+{
+    "status_code": "cdc-400",
+    "status_message": " alias is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 6
+
+Request Payload :
+```
+{
+    "parent_id": 2,
+    "alias": "Groups"
+}
+```
+ 
+Response HTTP Status Code : 400
+
+Response Payload:
+```
+{
+    "status_code": "cdc-400",
+    "status_message": " alias is already exist in acos table",
+    "data": null
+}
+```
+
+#### Case : Negative Case 7
+
+Request Payload :
+```
+{
+    "parent_id": 2,
+    "alias": "Groups:index"
+}
+```
+ 
+Response HTTP Status Code : 404
+
+Response Payload:
+```
+{
+    "status_code": "cdc-404",
+    "status_message": " parent_id not found in acos table",
+    "data": null
+}
+```
+#### Case : Positive Case
+
+Request Payload :
+```
+{
+    "parent_id": 2,
+    "alias": "Groups:index"
+
+}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "CDC-200",
+    "status_message": "OK",
+    "data": {
+        "id":3,
+        "parent_id": 2,
+        "alias": "Groups:index",
+        "created": "2020-10-28T08:58:13+00:00",
+        "modified": "2020-10-28T08:58:13+00:00"
+    }
+}
+```
+
+## <a name="delete"></a>DELETE
+
+### Endpoint
+DELETE /access/:id
+
+### Headers
+Key | Value 
+--- | ---
+Content-Type |*
+Accept | application/json
+Email | cdc_user@gmail.com
```

Comment:
also at DELETE /access/:id, it's doesn't email header. Because there is no filtering data in the endpoint. so remove email on header. 
You need email header just in : 
- GET /access
- POST /access/grant/:id
- POST /access/revoke/:id
thanks.